lib/web_ui/lib/src/engine/surface/picture.dart

```diff
@@ -285,7 +299,12 @@ class PersistedStandardPicture extends PersistedPicture {
 
       final bool fits = candidate.doesFitBounds(bounds);
       final bool isSmaller = candidatePixelCount < lastPixelCount;
-      if (fits && isSmaller) {
+      // [isTooSmall] is used to make sure that a small picture doesn't
+      // reuse and hold onto memory of a large canvas.
+      final bool isTooSmall = isSmaller &&
```

Comment:
Let's rename the variable to `isCanvasTooBig`. Otherwise, the distinction between `isSmaller` and `isTooSmall` is unclear.

Reply:
I would also move this computation inside the `if` block, because if the canvas doesn't fit or is not smaller that the previous canvas, then this computation has no effect (you'd also be able to remove the `isSmaller` from this expression).